### PR TITLE
[5.4] Add orderByDesc() to Query Builder

### PR DESCRIPTION
**Pull request by KennedyTedesco**
Just a shortcut of something that I use a lot.

### CODE REVIEW

**Comment by GrahamCampbell** [comment]
👍

**Comment by joshmanders** [comment]
So I'm on the fence about this... If we introduce `orderByDesc($name)` then shouldn't we have `orderByAsc($name)` too? Other wise inconsistencies... 

Then I think, maybe a magic method, that does like `orderByName('desc')` and `Name` is column name, sorta like magic method for `whereName('whatever')`.

Thoughts?

**Comment by KennedyTedesco** [comment]
I don't think so. `asc` it's already implicit (see orderBy) and, after all, this is the default of MySQL:

> The default sort order is ascending, with smallest values first. To sort in reverse (descending) order, add the DESC keyword. https://dev.mysql.com/doc/refman/5.7/en/sorting-rows.html

**Comment by stefensuhat** [comment]
actually Laravel is provide `latest` and `oldest`. you can pass the argument column name. ex: `latest('posted_at')`

**Comment by aliusa** [comment]
orderByDesc() is more remembable than latest()...

File: src/Illuminate/Database/Query/Builder.php [hunk 1]
```diff
@@ -1436,6 +1436,19 @@ public function orderBy($column, $direction = 'asc')
...
+     * Add a descending "order by" clause to the query.
+     *
+     * @param  string  $column
+     * @return $this
+     */
+    public function orderByDesc($column)
+    {
+        $this->orderBy($column, 'desc');
+
+        return $this;
```
Comment (CupOfTea696) [inline src/Illuminate/Database/Query/Builder.php]: Why not just use

```php
    public function orderByDesc($column)
    {
        return $this->orderBy($column, 'desc');
    }
```

Comment (KennedyTedesco) [inline src/Illuminate/Database/Query/Builder.php]: Here you go @CupOfTea696 https://github.com/laravel/framework/pull/18465

Thanks. 😄

Comment (anuppatil95) [inline src/Illuminate/Database/Query/Builder.php]: this would be perfect

Comment (CupOfTea696) [inline src/Illuminate/Database/Query/Builder.php]: Awesome!

**Pull request merged**